### DISCUSSION TITLE
Simpler help, justfile checking, formatting; standard prelude

### DISCUSSION THREAD

**Discussion by simonmichael** [casey/just#1765] [Ideas]
At the start of my justfiles I often put recipes like:
```
# Make constants and recipe arguments available as environment variables,
# (But {{ VAR }} handles multi-word values better.)
#set export := true

# By default, list this justfile's commands, optionally filtered by case-insensitive REGEX
@_help *REGEX:
    if [[ '{{ REGEX }}' =~ '' ]]; then just -ul; else just -ul --color=always | rg -i '{{ REGEX }}'; true; fi

alias h := _help

# Check this justfile for errors and non-standard format
@_chk:
    just --fmt --unstable --check

alias chk := _chk

# If this justfile is error free but in non-standard format, reformat it
@_fmt:
    just -q chk || just -q --fmt --unstable

alias fmt := _fmt
```

I don't have very concrete suggestions, just

1. I find the above recipes handy, others might too.
2. The ergonomics could be improved; look at all the verbosity required to smoothly check or format a justfile. I made these recipes because these command lines were quite hard to discover.
3. I appreciate `just` remaining flexible, yet things like recipe searching and easy justfile checking/formatting are such a common need, I wonder if such "prelude" recipes could be built in/easily available somehow.

**Comment by edvardm** [comment]
I like the idea! Linked this in [in my post](https://github.com/casey/just/discussions/2218) which you might find interesting as well

**Comment by michaelfortunato** [comment]
I am a fan of this idea as well!

**Comment by simonmichael** [comment]
(Cleaned up the example prelude a bit.)
